KarhooUISDK/Screens/CheckoutScreen/NewCheckout/Workers/KarhooNewCheckoutPaymentAndBookingWorker.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  KarhooNewCheckoutPaymentWorker.swift
+//  KarhooUISDK
+//
+//  Created by Aleksander Wedrychowski on 12/01/2023.
+//  Copyright © 2023 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+import KarhooSDK
+import UIKit
+import SwiftUI
+import Combine
+
+enum CheckoutBookingError {
+    case noUser
+    case cardAuthenticationFailure(message: String?)
+}
+
+/// This is only on concept. Feel free to adjust it if needed.
+enum BookingState {
+    case idle
+    case failure(CheckoutBookingError)
+    case success
+}
+
+protocol NewCheckoutPaymentAndBookingWorker: AnyObject {
+    var statePublisher : Published<BookingState>.Publisher { get }
+    func isReadyToPerformPayment() -> Bool
+    func performBooking()
+    func getPaymentNonce() -> Nonce?
+}
+
+final class KarhooNewCheckoutPaymentAndBookingWorker: NewCheckoutPaymentAndBookingWorker {
+
+    // MARK: - Depencencies
+
+    private let userService: UserService
+    private let sdkConfiguration: KarhooUISDKConfiguration
+    private let analytics: Analytics
+
+    // MARK: - Properties
+
+    private var quote: Quote!
+    private(set) var paymentNonce: Nonce?
+    private(set) var loyaltyNonce: LoyaltyNonce?
+    private(set) var passengerDetails: PassengerDetails?
+    private(set) var bookingRequestInProgress = false
```

Comment:
You introduced `enum BookingState` so maybe `inProgress` should be one of states, instead of this "`private(set) var bookingRequestInProgress = false`? It's also kind of status.